File: search/linear_search/haskell/linear_search.hs
module LinearSearch where
    
    linear_search::[Integer]->Integer->Integer
    linear_search l x= search l x 0

    search::[Integer]->Integer->Integer->Integer
    search (x:xs) y n = if x == y then n
                                else search xs y (n + 1)
    search [] y n = -1

File: search/linear_search/golang/Linear_search.go
// Linear Search in Golang

package main

import "fmt"

func linearsearch(datalist []int, key int) bool {
    for _, item := range datalist {
        if item == key {
            return true
        }
    }
    return false
}
func main() {
    items := []int{95,78,46,58,45,86,99,251,320}
    fmt.Println(linearsearch(items,58))
}
